rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -1222,6 +1221,35 @@ public JobId copyJobAndResubmitWithGeneralInfo(JobId jobId, Map<String, String>
         return null;
     }
 
+    @Override
+    public Job getJobContent(JobId jobId) throws NotConnectedException, PermissionException, UnknownJobException,
+            JobCreationException, SubmissionClosedException {
+        File tempFile = null;
+        try {
+            final InputStream jobContent = restApi().getJobContent(sid, jobId.value());
+            tempFile = File.createTempFile("jobContent", jobId.value());
+            FileUtils.copyInputStreamToFile(jobContent, tempFile);
+            final Job job = JobFactory.getFactory().createJob(tempFile.getPath());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "job". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)